code/modules/procedural_mapping/mapGeneratorModules/helpers.dm

```diff
@@ -10,12 +10,12 @@
 	if(!mother)
 		return
 	var/list/map = mother.map
-	for(var/turf/T in map)
-		SSair.remove_from_active(T)
-	for(var/turf/open/T in map)
-		if(T.air)
-			T.air = T.create_gas_mixture()
-		SSair.add_to_active(T, TRUE)
```

Comment:
I get the feeling that the issue here was the add to active, which is required and this avoids.
I believe the runtime you saw in process_cell was an inherent flaw in the logic, and not a bug caused by making space turfs non space, it isn't good to mask that.

No idea about the fucky gas tho, the blocks_air may be helpful here. honestly create_gas_mixture should be a valid call on space turfs, if it isn't then we've fucked up. shouldn't need to inline it like this to fix that fuckup